File: app/src/main/java/com/ssowens/android/myweatherapp/service/WeatherApi.java
package com.ssowens.android.myweatherapp.service;

import com.ssowens.android.myweatherapp.model.WeatherForecast;
import com.ssowens.android.myweatherapp.model.WeatherResponseByCity;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

/**
 * Created by Sheila Owens on 2/25/19.
 */
public interface WeatherApi {

    @GET("data/2.5/weather?")
    Call<WeatherResponseByCity> getCurrentWeatherData(
            @Query("lat") String lat,
            @Query("lon") String lon,
            @Query("units") String units,
            @Query("APPID") String app_id
    );

    @GET("data/2.5/weather?")
    Call<WeatherResponseByCity> getCurrentWeatherById(
            @Query("id") String id,
            @Query("units") String units,
            @Query("APPID") String app_id
    );

    @GET("data/2.5/forecast?")
    Call<WeatherForecast> getForecast(
            @Query("lat") String lat,
            @Query("lon") String lon,
            @Query("units") String units,
            @Query("APPID") String app_id
    );
}

File: app/src/main/java/com/ssowens/android/myweatherapp/service/WeatherApi.kt
package com.ssowens.android.myweatherapp.service

import com.ssowens.android.myweatherapp.model.Weather
import kotlinx.coroutines.Deferred
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Path

/**
 * Created by Sheila Owens on 2/23/19.
 */
// A retrofit Network Interface for the Api
interface WeatherApi {
  @GET("weather/{id}")
  fun getWeatherByCity(@Path("id") id: Int): Deferred<Response<Weather>>
}